File: proposed-projects/round-trip-experiment/reverse/20260909T153339Z_9de9fd66/20260910T090338Z_5eab4940/ported_implementation/src/grammar-graph/get-input-as-code-points.ts
import type { ValidInput } from './types.js';

export const getInputAsCodePoints = (src: ValidInput): number[] => {
  if (typeof src !== 'string') {
    return Array.isArray(src) ? src : [src];
  }

  return [...src].map((char) => char.codePointAt(0)!);
};

File: packages/gbnf-experiment/data/20260909T194429Z_a7bb3ecf/ported_implementation/grammar_graph/get_input_as_code_points.py
from __future__ import annotations

from typing import List

from .types import ValidInput


def get_input_as_code_points(src: ValidInput) -> List[int]:
    if not isinstance(src, str):
        return list(src) if isinstance(src, (list, tuple)) else [src]

    return [ord(char) for char in src]
